---
title: "Transcriptomics"
author: "Jacob Gerzenshtein"
date: "May 17, 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Working on Transcriptomics 

Importing the data

```{r}
metadata <- read.csv("airway_metadata.csv", stringsAsFactors = FALSE)
counts <- read.csv("airway_scaledcounts.csv", stringsAsFactors = FALSE)

head(counts)

head(metadata)
```

Viewing data

```{r}
View(metadata)
```

Finding data points that are control, then calculating mean counts per gene

```{r}
control <- metadata[metadata[ ,"dex"] == "control",]
control.mean <- rowSums(counts[ ,control$id]) / nrow(control) 
names(control.mean) <- counts$ensgene
```


To make the code above more robust, divide by generic number of controls

```{r}
treated <- metadata[metadata[ ,"dex"] == "treated",]
treated.mean <- rowSums(counts[ ,treated$id]) / nrow(treated)
names(treated.mean) <- counts$ensgene
```


Combining the two 

```{r}
meancounts <- data.frame(control.mean, treated.mean)

meancounts
```

Finding the column data

```{r}
colSums(meancounts)
```


Creating a scatter plot

```{r}
plot(meancounts, log = "xy")
```

Using log2 to look at differentially expressed genes

```{r}
meancounts$log2fc <- log2(meancounts[ ,"treated.mean"] / meancounts[ ,"control.mean"])
head(meancounts)
```

Filtering out data to get rid of NaN and -inf 

```{r}
zero.vals <- which(meancounts[ ,1:2] == 0, arr.ind = TRUE)

to.rm <- unique(zero.vals[ ,1])

mycounts <- meancounts[-to.rm, ]

head(mycounts)

View(mycounts)
```

The purpose of the arr.ind argument: creates an array index when x is an array, will not return the value it will return the location of the desired data instead

Need to use unique to get rid of duplicate elements/rows


Finding number of genes that are up or down regulated 

```{r}
up.ind <- mycounts$log2fc > 2

down.ind <- mycounts$log2fc < (-2)

paste("The amount of up is:", sum(up.ind))

paste("The amount of down is:", sum(down.ind))
```


Introducing annotation to the data 

```{r}
anno <- read.csv("annotables_grch38.csv")

head(anno)
View(anno)
```


Merging the annotation with the mycounts data

```{r}
results <- merge(mycounts, anno, by.x = "row.names", by.y = "ensgene")
View(results)
```


Loading annotation packages

```{r}
#biocLite("org.Hs.eg.db")
library("AnnotationDbi")
library("org.Hs.eg.db")
```


Adding individual columns to results 

```{r}
mycounts$symbol <- mapIds(org.Hs.eg.db, keys = row.names(mycounts),
                          column = "SYMBOL",
                          keytype = "ENSEMBL",
                          multiVals = "first")
```
```{r}
mycounts$entrez <- mapIds(org.Hs.eg.db, keys = row.names(mycounts),
                          column = "ENTREZID",
                          keytype = "ENSEMBL",
                          multiVals = "first")

mycounts$uniprot <- mapIds(org.Hs.eg.db, keys = row.names(mycounts),
                          column = "UNIPROT",
                          keytype = "ENSEMBL",
                          multiVals = "first")
```



```{r}
head(mycounts)
```


Examining results

```{r}
head(mycounts[up.ind, ])
```



 